### PR TITLE
IOS only upgrade react-native to 0.72.x

### PR DESCRIPTION
**Pull request by siddarthkay**
This PR is just made to test and fix effects of this upgrade on CI.
Note: This PR only contains IOS side of the upgrades.
state: Ready

### CODE REVIEW

File: nix/deps/nodejs-patched/default.nix [hunk 1]
```diff
@@ -49,12 +47,7 @@ stdenv.mkDerivation {
       ${patchMavenSources} $modBuildGradle
     done
   '';
-  # Do not add a BuildId to the generated libraries, for reproducibility
-  patchBuildIdPhase = ''
-    substituteInPlace ./node_modules/react-native/ReactAndroid/src/main/jni/Application.mk --replace \
```
Comment (siddarthkay) [inline nix/deps/nodejs-patched/default.nix:54]: now that `ReactAndroid/src/main/jni/Application.mk` no longer exists, We'll have to figure out where to update build-id :)

**Comment by siddarthkay** [comment]
Hit a blocker at `react-native` version `0.71.12`

Although builds are green at the moment but the app indefinitely stays on the splash screen and when you put the app in background and open it again you are greeted with this not so specific error.

Simulator Screenshot - iPhone 11 Pro - 2023-07-20 at 09 32 58

My current hypothesis is that I need to more thoroughly clean up all of the v1 usage of `react-native-redash` library which in turn uses the v1 of `react-native-reanimated` library.

More updates on this later.

**Comment by siddarthkay** [comment]
I briefly had this error after the commit for re-animated upgrade.

Screenshot_2023-08-03_at_7 02 46_PM

**Comment by siddarthkay** [comment]
However I am back with my old nemesis and now this is now the latest error.
It seems to be reagent related. 
I hate how this does not give out any valuable information.

Screenshot 2023-08-13 at 6 34 17 PM

**Comment by siddarthkay** [comment]
IOS Side is ready, time to focus on Android side

Screenshot 2023-08-19 at 9 39 56 AM

**Comment by flexsurfer** [comment]
looks stale, feel free to reopen if needed

**Pull request closed without merging**